### DISCUSSION TITLE
How to invert gimbal direction?

### DISCUSSION THREAD

**Discussion by skaman82** [EdgeTX/edgetx#5600] [Q&A]
I just installed the new AG01 NANO in my Jumper T-Pro (V1). I installed the gimbals exactly in the same position as the previous gimbals. In EdgeTX the gimbals are now exactly reversed. How can I reverse the direction - only found the channel inversion in the outputs menu - but that way I constantly get a throttle warning on powering up the radio. I can't find a setting under "Hardware" to invert the gimbal direction - this works only for pots. Have I overlooked something? Unfortunately I can't simply turn the gimbals 180 deg.

**Comment by pfeerick** [comment]
Normally if doing a non-standard mod, you would fix the hardware itself, and swap the pos. and neg. for the gimbal wires over. However, since these are hall gimbals, it may be polarity sensitive so I wouldn't recommend that until knowing otherwise. 

So you probably need to do a custom build of the firmware, while changing the ADC direction for those inputs as needed (first 4 should be the gimbals):

https://github.com/EdgeTX/edgetx/blob/fcc75bd6b74293f5cdb1802dbed94e335e2b0344/radio/src/targets/taranis/hal.h#L1899-L1900

**Reply by skaman82** [reply]
Thants for pointing me in the right direction. I modified the code and made a custom build. Now the gimbals are working correctly :)
